### ISSUE TITLE
Incorrect fill tessellation?

### ISSUE THREAD

**Issue by tomfclarke** [nical/lyon#486]
This library is awesome, but I may have found some unexpected behaviour.

I've managed to reproduce this behaviour in the glium_basic example by replacing the `build_logo_path()` call with some commands to build a custom path.

This path draws how I would expect:
```
    //build_logo_path(&mut builder);
    builder.move_to (point( 0.1, 15.418));
    builder.line_to (point(38.6, 15.399));
    builder.line_to (point(45.0, 15.398)); // if this point's y-value is equal to or lower than the one before, the shape doesn't seem to be drawn properly
    builder.line_to (point(64.1, 15.395));
    builder.line_to (point(12.8, 13.101));
    builder.line_to (point( 6.4, -0.139));
    builder.close();
```
<img width="140" alt="correct-tessellation" src="https://user-images.githubusercontent.com/7001196/61729059-5b394b00-ad6e-11e9-9fbf-3cefc76086c3.png">

But if the y-value of the point in question is changed to equal the one before it:
```
    //build_logo_path(&mut builder);
    builder.move_to (point( 0.1, 15.418));
    builder.line_to (point(38.6, 15.399));
    builder.line_to (point(45.0, 15.399)); // this point's y-value now equals the one before
    builder.line_to (point(64.1, 15.395));
    builder.line_to (point(12.8, 13.101));
    builder.line_to (point( 6.4, -0.139));
    builder.close();
```
Then the output is as below (when I expected it to be the same):
<img width="143" alt="incorrect-tesselation" src="https://user-images.githubusercontent.com/7001196/61729304-d0a51b80-ad6e-11e9-964f-3242c90cee1e.png">

With other path drawing libraries the output remains the same as the first example.

**Comment by nical** [comment]
Thanks for the test case! I'll  have a look soon.

**Comment by nical** [comment]
@tomfclarke thanks again for reporting this. The bug was quite subtle and your minimal testcase was super helpful.

I published a fix in version [0.14.1](https://crates.io/crates/lyon/0.14.1) and back-ported it to the 0.13.x branch in version [0.13.5](https://crates.io/crates/lyon/0.13.5) since I suspect most people are still on 0.13.x (ggez for example, cc @icefoxen).

**Comment by tomfclarke** [comment]
No problem @nical. Thanks for fixing it so quickly!
